src/context/scan.js

```diff
@@ -0,0 +1,71 @@
+/**
```

Comment:
These functions are only being used in `test-node-discover.js`. Should they live within that file (or at least within an obviously test file).  Although it makes sense to leave her if you believe these helpers will soon be used by other non-test files.

Reply:
They will be used here: https://github.com/ampproject/amphtml/pull/29197. I just wanted to have this pull request to have all semantics for node (parent/child) navigation/discovery. So, these utils are a bit ahead of their time.